packages/fxa-auth-server/lib/luaScripts/getSessionTokens_1.lua

```diff
@@ -0,0 +1,58 @@
+-- https://github.com/mozilla/fxa-auth-server/pull/2254 for packing scheme info
+local REDIS_SESSION_TOKEN_PROPERTIES = {
+  'lastAccessTime',
+  'location',
+  'uaBrowser',
+  'uaBrowserVersion',
+  'uaOS',
+  'uaOSVersion',
+  'uaDeviceType',
+  'uaFormFactor'
+}
+local REDIS_SESSION_TOKEN_LOCATION_PROPERTIES = {
+  'city',
+  'state',
+  'stateCode',
+  'country',
+  'countryCode'
+}
+
+local function unpackArray(keys, values)
+  local result = {}
+  for i, k in ipairs(keys) do
+    result[k] = values[i]
+  end
+  return result
+end
+
+local function unpackSessions(value)
+  if not value then
+    return {}
+  end
+  local result = {}
+  local tokens = cjson.decode(value)
+  for id, values in pairs(tokens) do
+    if type(values) == 'table' and values[1] then
+      -- it's an array
+      local token = unpackArray(REDIS_SESSION_TOKEN_PROPERTIES, values)
```

Comment:
By moving to lua scripts would we no longer have to `truncatePackedArray`?

Reply:
good question! By virtue of `JSON.stringify` keys with `undefined` values get eliminated. Here, missing keys will be replaced by `null` if they are leading and excluded if they're trailing. For example
```
...
"location": {"state":"FL"},
...
```
gets encoded to
```
... [null,"WA"] ...
```
because city was undefined, along with the rest

But we also ought to remove `"uaFormFactor":null` so I've updated the stringify in redis.js to:
```js
// remove keys with null values
const json = JSON.stringify(token, (k,v) =>(v == null ? undefined : v))
```